grading_algorithm.py

```diff
@@ -0,0 +1,163 @@
+    position_percent = ((total_word_count - position_error_count) / total_word_count) * 100
+    position_percent_of_order_score = position_percent * .2
+
+    # Adds to total
+    total_order_score = total_order_score + position_percent_of_order_score
+    # Word count check 
+    if word_count(expression, c_exp) == True:
+        # Add 10% to the overrall order score if the word count matches
+        total_order_score = total_order_score + 10
+
+    # Performs the word order check
+    order_error_count = word_order(expression, c_exp)
+    
+    # Once again converts to percent, but this section is instead worth 70% of the order portion
+    order_percent = ((total_word_count - order_error_count) / total_word_count) * 100
+    order_percent_of_total = order_percent * .7
+    total_order_score = total_order_score + order_percent_of_total
+
+    # The presence check
+    presence_error_count = word_prescence(expression, c_exp)
+
+    # Converts presence score to a percent which is the overrall score for the presence portion
+    presence_percent = ((total_word_count - presence_error_count) / total_word_count) 
+    presence_score = presence_percent * 100
+
+    # Combines the two sections, both being weighted at 50%
+    grade = (total_order_score * .5) + (presence_score * .5)
+    return grade
+
+def correct_expression(expression, c_exp):
+    """
+    This is a simple function that determines if the expressions are the
+    Args:
+        expression (string): The student provided version of the expression
+        c_exp (string): The correct expression to be comapred to
+    
+    Returns:
+        bool: simple true/false return for if the expression is correct
+    """
+    if expression == c_exp:
+        return True
+    return False
+
+def word_prescence(expression, c_exp):
+    """
+    This is a function that determines if the desired words are present within the sentence
+    Args:
+        expression (string): The student provided version of the expression
+        c_exp (string): The correct expression to be comapred to
+    
+    Returns:
+        error_count(int): the amount of times that a word is not present
+    """
+    c_exp_list = c_exp.split()
+    error_count = 0
+
+    #Very simple checks if all correct words are present
+    for x in range(len(c_exp_list)):
+        if c_exp_list[x] not in expression:
+            error_count+=1
+
+    return error_count
+
+def word_order(expression, c_exp):
+    """
+    This is a function that determines if there is an error with the order of words
+    Args:
+        expression (string): The student provided version of the expression
+        c_exp (string): The correct expression to be comapred to
+    
+    Returns:
+        error_count (int): the amount of times that a word order error occurs
+    """
+    expression_list = expression.split()
+    c_exp_list = c_exp.split()
+    error_count = 0
+    for x in range(len(expression_list)):

```

Comment:
I think it would provide less overhead and look a bit nicer if you looped til len(exp_list) -1 rather than len(exp_list). This way you can remove that check on line 102. You can also then just move the last word check out of the loop to perform those final operations.